app/controllers/books_controller.rb

```diff
@@ -10,4 +10,34 @@ def show
       redirect_to books_path
     end
   end
-end
\ No newline at end of file
+
+  def new
+    @book = Book.new
+  end
+
+  def create
+    book = Book.new(
+      year_published: book_params[:year_published],
+      page_count: book_params[:page_count],
+      thumbnail: book_params[:thumbnail].presence || 'https://nnp.wustl.edu/img/bookCovers/genericBookCover.jpg',
+      title: book_params[:title].titlecase)
+    if book.save
+      author_input = book_params[:authors].split(",")
+
+      author_input.each do |author|
+        book.authors << Author.find_or_create_by(name: author.titlecase.strip)
+      end
+
+      redirect_to book_path(book)
+    else
+      flash[:notice] = "This book has already been created"
```

Comment:
Hm. I'm not sure this code is actually accomplishing what the user story says. I don't see any check that the book title is unique. I think the `else` here is just that the book failed to be created (for any reason), so that's probably what the flash should say. @DavisC0801

Reply:
Added new flash edge case for general book creation failure